### PR TITLE
Remove FlutterApplication from app templates.

### PR DESCRIPTION
**Pull request by hpoul**
## Description

There is an inconsistency between the migration guide for pre flutter 1.12 android applications: https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects and the current app templates.

According to the guide (for both, *Full-Flutter app migration* as well as for *Add-to-app migration*):

> 3. Remove the reference to FlutterApplication from the application tag.

This PR removes `FlutterApplication` from `AndroidManifest.xml` to make it consistent with manually upgraded projects.

## Related Issues

* No issue which would be fixed

## Tests

No tests, as this does not add any functionality.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

**Comment by xster** [comment]
Reminds me I need to merge https://github.com/flutter/engine/pull/20769. Let's merge this after our next branch cut next week.

**Review by xster — APPROVED** [review]
This LGTM. flutter/engine#20769 is merged as well. 

Please wait until Sept 14th before merging this.

**Comment by jonahwilliams** [comment]
@xster September 14th of this year, right? :)

**Comment by xster** [comment]
Thanks for reminding Jonah!

**Pull request merged**
